hudi-common/src/test/java/org/apache/hudi/common/table/read/TestUpdateProcessor.java

```diff
@@ -145,7 +146,7 @@ void testHandleInsertWithPayload(boolean shouldIgnore) {
     when(recordContext.seal(any())).thenAnswer(invocationOnMock -> invocationOnMock.getArgument(0));
     BufferedRecord<IndexedRecord> previous = null;
     BufferedRecord<IndexedRecord> merged = getRecord("value2", null);
-    BufferedRecord<IndexedRecord> expected = getRecord("value2", HoodieOperation.INSERT);
+    BufferedRecord<IndexedRecord> expected = getRecordWithSerializableIndexedRecord("value2", HoodieOperation.INSERT);
```

Comment:
we can also chenge the `getRecord` to always create `SerializableIndexRecord` there, because that is on par with the actual behavior: the buffer records merging do not trigger avro deserailization.

Reply:
sure.